### ISSUE TITLE
Why only async version TryUpdateModelAsync exist?

### ISSUE THREAD

**Issue by justserega** [aspnet/Mvc#5538]
I do not want use async/await pattern in some places, why there are no TryUpdateModel method?

**Comment by Eilon** [comment]
The underlying model binding system is async because it often has to do operations such as reading data from places that offer async APIs. The thing with async is that it's "viral" - once you use it somewhere deep down in a system, you need to be async all the way out.

Is there a reason you don't want to use async/await?

**Comment by justserega** [comment]
> Is there a reason you don't want to use async/await?

Because it is "viral". If all my code not async, it is little strange do async controller only for UpdateModel.  Is there a real need to make such light method async? In addition there is some overhead on thread context switching, is async in this place really needed?

**Comment by Eilon** [comment]
@rynowak any thoughts on this? Have we seen any noticeable perf differences between async actions or not? (I'm expecting that any such differences would get drowned out entirely in any meaningful scenario.)

**Comment by rynowak** [comment]
There's a pretty good blog article I came across on this topic this week: http://www.jeremykruer.com/why-my-synchronous-api-scaled-better-than-my-asynchronous-api/

TLDR summary: async allows you scale better than sync. It is not any faster for serving a single request, but it is much more efficient for serving many requests.

**Comment by Eilon** [comment]
We are closing this issue because no further action is planned for this issue. If you still have any issues or questions, please log a new issue with any additional details that you have.
